x-pack/platform/plugins/shared/fleet/server/services/agent_policies/package_policies_to_agent_inputs.ts

```diff
@@ -197,3 +199,20 @@ const globalDataTagsToAddFields = (tags: GlobalDataTag[]): FullAgentPolicyAddFie
     },
   };
 };
+
+const filterGlobalDataTags = (
+  globalDataTags: GlobalDataTag[] | undefined,
+  packageInfo: PackageInfo | undefined
+): GlobalDataTag[] | undefined => {
+  const agentlessGlobalDataTags = getAgentlessGlobalDataTags(packageInfo);
+
+  if (!globalDataTags || !packageInfo || !agentlessGlobalDataTags) {
```

Comment:
nit: could probably be simplified to just this if I'm following correctly:
```suggestion
  if (!agentlessGlobalDataTags) {
```